### PR TITLE
Validate m2m relation fields do not contain referential actions

### PR DESCRIPTION
**Pull request by tomhoule**
They are not supported and are no-ops, we should reject them.

relevant:  prisma/prisma#10861

### CODE REVIEW

File: libs/datamodel/core/tests/attributes/relations/many_to_many_negative.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+use expect_test::expect;
+
+#[test]
+fn many_to_many_relation_fields_with_referential_actions() {
+    let schema = r#"
```
Comment (pimeys) [inline libs/datamodel/core/tests/attributes/relations/many_to_many_negative.rs:5]: iiindoooc.

File: libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/relations/many_to_many.rs [hunk 2]
```diff
@@ -37,3 +37,22 @@ pub(crate) fn validate_singular_id(relation: ImplicitManyToManyRelationWalker<'_
         }
     }
 }
+
+pub(crate) fn validate_no_referential_actions(
+    relation: ImplicitManyToManyRelationWalker<'_, '_>,
+    ctx: &mut Context<'_>,
+) {
+    let referential_action_args = [relation.field_a(), relation.field_b()].into_iter().flat_map(|field| {
```
Comment (pimeys) [inline libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations/relations/many_to_many.rs:45]: maybe `referential_action_spans`?

File: migration-engine/core/src/rpc.rs [hunk 3]
```diff
@@ -1,5 +1,7 @@
 use crate::{CoreError, CoreResult, GenericApi};
-use jsonrpc_core::{types::error::Error as JsonRpcError, IoHandler, Params};
+#[cfg(any(feature = "sql", feature = "mongodb"))]
+use jsonrpc_core::IoHandler;
+use jsonrpc_core::{types::error::Error as JsonRpcError, Params};
```
Comment (pimeys) [inline migration-engine/core/src/rpc.rs:4]: under this some wasm warnings.

File: libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations.rs [hunk 4]
```diff
@@ -148,6 +148,7 @@ pub(super) fn validate(ctx: &mut Context<'_>, relation_transformation_enabled: b
             }
             RefinedRelationWalker::ImplicitManyToMany(relation) => {
                 relations::many_to_many::validate_singular_id(relation, ctx);
+                relations::many_to_many::validate_no_referential_actions(relation, ctx);
```
Comment (pimeys) [inline libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline/validations.rs:151]: I like how simple this is nowadays 💯 ..

**Review by pimeys — APPROVED** [review]

File: migration-engine/core/src/rpc.rs (hunk 3 above)
Comment (tomhoule) [reply migration-engine/core/src/rpc.rs:4]: ugh, all these changes are here to get rid of the warnings. I give up.

**Pull request merged**